akka-remote/src/main/scala/akka/remote/artery/FixedSizePartitionHub.scala

```diff
@@ -0,0 +1,70 @@
+/**
+ * Copyright (C) 2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote.artery
+
+import akka.annotation.InternalApi
+import akka.stream.scaladsl.PartitionHub
+import org.agrona.concurrent.OneToOneConcurrentArrayQueue
+import java.util.concurrent.atomic.AtomicInteger
+import org.agrona.concurrent.ManyToManyConcurrentArrayQueue
+
+/**
+ * INTERNAL API
+ */
+@InternalApi private[akka] class FixedSizePartitionHub[T](
```

Comment:
Could this be general purpose in streams instead of custom special stage in artery?